cartography/graph/sanitizer.py

```
@@ -0,0 +1,135 @@
+
+    if node_schema.sub_resource_relationship:
+        for p_ref in asdict(node_schema.sub_resource_relationship.properties).values():
+            results[p_ref.name] = p_ref
+        for p_ref in asdict(
+            node_schema.sub_resource_relationship.target_node_matcher
+        ).values():
+            results[p_ref.name] = p_ref
+
+    if node_schema.other_relationships:
+        for rel in node_schema.other_relationships.rels:
+            for p_ref in asdict(rel.properties).values():
+                results[p_ref.name] = p_ref
+            for p_ref in asdict(rel.target_node_matcher).values():
+                results[p_ref.name] = p_ref
+
+    return results
+
+
+def _auto_format_field(p_ref: PropertyRef, value: Any) -> Any:
+    """This function auto-formats a field based on the PropertyRef's auto_format type."""
+    try:
+        if value is None:
+            return None
+        if p_ref.auto_format is None:
+            return value
+        if p_ref.auto_format == str:
+            value = str(value)
+            if len(value) == 0:
+                return None
+            return value
+        if p_ref.auto_format == int:
+            return int(value)
+        if p_ref.auto_format == float:
+            return float(value)
+        if p_ref.auto_format == dict or p_ref.auto_format == list:
+            if len(value) == 0:
+                return None
+            return value
+        if p_ref.auto_format == datetime:
+            if isinstance(value, datetime):
+                return value
+            if isinstance(value, int) or isinstance(value, float):
+                return datetime.fromtimestamp(value)
+            return parse_date(value)
+        if p_ref.auto_format == bool:
+            if isinstance(value, str):
+                if value.lower() in ("true", "1", "yes"):
+                    return True
+                if value.lower() in ("false", "0", "no"):
+                    return False
+                return value
```

Thoughts on logging a warning here if we can't convert the value to a boolean? All other paths within this if statement return a boolean whereas this path silently returns a string. I see our default is to return a string but I think it may be nice to warn because if a string is being returned there is likely some unexpected output coming from the source API

---

Good catch.